src/RA_StringUtils.cpp

```diff
@@ -85,14 +85,16 @@ std::string Narrow(const std::string& str)
 _Use_decl_annotations_
 std::string& TrimLineEnding(std::string& str) noexcept
 {
-    auto nIndex = str.length();
-    if (nIndex > 0 && str[nIndex - 1] == '\n')
-        nIndex--;
-    if (nIndex > 0 && str[nIndex - 1] == '\r')
-        nIndex--;
-
-    if (nIndex != str.length())
-        str.resize(nIndex);
+    auto dCount = std::count_if(str.begin(), str.end(), [](const char c) noexcept
+    {
+        return (c == '\r' || c == '\n');
+    });
+
+    while (dCount != 0)
+    {
+        str.pop_back();
+        dCount--;
+    }
```

Comment:
I could have made this logarithmic but because some of tests had line endings in the middle it couldn't be helped. It's still pretty fast.

```
Test                      | Time
‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
TestTrimLineEnding        | 0:00:00.0001801
TestThreeLinesEmptyMiddle | 0:00:00.0001469
TestOneLineWithLineEnding | 0:00:00.00005
TestTwoLines              | 0:00:00.0001768
```
The way it was before made the tests fail after correcting the `TestTrimLineEnding` test.